File: sdk/java/src/main/java/io/apibrew/client/model/Token.java
package io.apibrew.client.model;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.Date;

public class Token {

    public static class Container {
        private Token token;

        public Token getToken() {
            return token;
        }

        public void setToken(Token token) {
            this.token = token;
        }
    }

    public enum TokenTerm {
        // 1 minute
        VERY_SHORT,
        // 20 minutes
        SHORT,
        // 2 days
        MIDDLE,
        // 60 days
        LONG,
        // 2 years
        VERY_LONG,
    }

    @JsonProperty
    private TokenTerm term;

    @JsonProperty
    private String content;

    @JsonProperty
    private Date expiration;
    public Date getExpiration() {
        return expiration;
    }

    public void setExpiration(Date expiration) {
        this.expiration = expiration;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public TokenTerm getTerm() {
        return term;
    }

    public void setTerm(TokenTerm term) {
        this.term = term;
    }
}

File: sdk/nodejs/src/model/token.ts
export enum TokenTerm {
    // 1 minute
    VERY_SHORT = 'VERY_SHORT',
    // 20 minutes
    SHORT = 'SHORT',
    // 2 days
    MIDDLE = 'MIDDLE',
    // 60 days
    LONG = 'LONG',
    // 2 years
    VERY_LONG = 'VERY_LONG',
}

export interface Token {
    term: TokenTerm;
    content: string;
}

export interface AuthenticationRequest {
    username: string;
    password: string;
    term: TokenTerm;
}

export interface AuthenticationResponse {
    token: Token;
}
